mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/impl/PojoIndexedTypeManager.java

```diff
@@ -135,7 +135,7 @@ void resolveEntitiesToReindex(PojoReindexingCollector collector, PojoRuntimeIntr
 		);
 	}
 
-	<R, O> MappedIndexSearchScopeBuilder<R, O> createSearchScopeBuilder(MappingContextImplementor mappingContext) {
+	<R, E2> MappedIndexSearchScopeBuilder<R, E2> createSearchScopeBuilder(MappingContextImplementor mappingContext) {
```

Comment:
Surely I did not get something. But why E2 instead of E?

Reply:
`E` is the type of targeted entities. `E2`, here, is the type returned when loading an entity.

For the ORM mapper, they are the same.

For the JavaBean mapper, E2 is `Void` because loading is not supported.

That's already enought to justify using a separate generic type parameter here. But going further, other mappers may use a container type for `E2`, such as `Entity<E2>`, that exposes additional method (`getId()`) or provides lazy loading.

Reply:
makes sense thanks